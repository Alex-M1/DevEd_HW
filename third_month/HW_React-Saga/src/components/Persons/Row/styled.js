import styled from 'styled-components';

export const StRow = styled.div`
  display: grid;
  grid-template-columns: repeat(5,1fr);
  position: relative;
`;

export const StCell = styled.div`
  border: 1px solid black;
  padding: 3px 0;
`;

export const StDelete = styled.span`
  position: absolute;
  right: -20px;
  font-size: 25px;
  cursor: pointer;
  align-self: center;
`;
