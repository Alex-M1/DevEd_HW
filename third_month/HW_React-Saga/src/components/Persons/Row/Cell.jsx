import React from 'react';
import PropTypes from 'prop-types';
import { StCell } from './styled';

function Cell({ id, type, children, onDoubleClick, updatePerson }) {
  const onBlur = (e) => {
    e.target.contentEditable = false;
    updatePerson({ id, type, value: e.target.textContent });
  };
  return <StCell onBlur={onBlur} onDoubleClick={onDoubleClick}>{children}</StCell>;
}

export default Cell;

Cell.propTypes = {
  id: PropTypes.string,
  type: PropTypes.string,
  children: PropTypes.string,
  updatePerson: PropTypes.func,
  onDoubleClick: PropTypes.func,
};
