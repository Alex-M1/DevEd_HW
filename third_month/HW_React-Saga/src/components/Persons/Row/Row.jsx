import React from 'react';
import PropTypes from 'prop-types';
import { StRow, StDelete } from './styled';
import Cell from './Cell';

function Row({ isHead, _id, name, age, phone, email, company, updatePerson, deletePerson }) {
  const onDblClick = (e) => {
    if (!isHead) {
      e.target.contentEditable = true;
    }
  };
  const onDeleteClick = () => deletePerson(_id);
  return (
    <StRow >
      <Cell onDblClick={onDblClick} updatePerson={updatePerson} id={_id} type="name">{name}</Cell>
      <Cell onDoubleClick={onDblClick} updatePerson={updatePerson} id={_id} type="age">{age}</Cell>
      <Cell onDoubleClick={onDblClick} updatePerson={updatePerson} id={_id} type="phone">{phone}</Cell>
      <Cell onDoubleClick={onDblClick} updatePerson={updatePerson} id={_id} type="email">{email}</Cell>
      <Cell onDoubleClick={onDblClick} updatePerson={updatePerson} id={_id} type="company">{company}</Cell>
      {!isHead ? <StDelete onClick={onDeleteClick}>&times;</StDelete> : null}
    </StRow>
  );
}

export default Row;

Row.propTypes = {
  _id: PropTypes.string,
  isHead: PropTypes.bool,
  name: PropTypes.string,
  age: PropTypes.string,
  phone: PropTypes.string,
  email: PropTypes.string,
  company: PropTypes.string,
  updatePerson: PropTypes.func,
  deletePerson: PropTypes.func,
};
