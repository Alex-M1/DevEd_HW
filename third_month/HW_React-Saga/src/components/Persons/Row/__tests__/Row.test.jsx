import { jest } from '@jest/globals';
import React from 'react';
import Row from '../Row';

describe('Row', () => {
  const props = {
    isHead: false,
    _id: 'mockValue',
    name: 'mockValue',
    age: 'mockValue',
    phone: 'mockValue',
    email: 'mockValue',
    company: 'mockValue',
    updatePerson: jest.fn(),
    deletePerson: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = global.shallowSmart(<Row {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StRow', () => {
    const component = global.mountSmart(<Row {...props} />);
    expect(component.find('styled__StRow')).toHaveLength(1);
  });
  it('Should have Cell', () => {
    const component = global.mountSmart(<Row {...props} />);
    expect(component.find('Cell')).toHaveLength(5);
  });
  it('Should have StDelete', () => {
    const component = global.mountSmart(<Row {...props} />);
    expect(component.find('styled__StDelete')).toHaveLength(1);
  });
  it('Should have StDelete click', () => {
    const component = global.mountSmart(<Row {...props} />);
    const item = component.find('styled__StDelete');
    item.simulate('click');
    expect(component.props().deletePerson).toHaveBeenCalledWith(props._id);
  });
  // it('Should have call blur & call updatePerson', () => {
  //   const newValue = 'Mock';
  //   const component = global.mountSmart(<Cell {...props} />);
  //   const item = component.find('styled__StCell');
  //   item.simulate('blur', { target: { textContent: newValue } });
  //   expect(component.props().updatePerson).toHaveBeenCalledWith({
  //     id: props.id,
  //     type: props.type,
  //     value: newValue,
  //   });
  // });
});
