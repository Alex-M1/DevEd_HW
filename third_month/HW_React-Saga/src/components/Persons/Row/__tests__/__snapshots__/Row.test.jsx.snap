// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Should match snapshot 1`] = `"<div class=\\"styled__StRow-sc-1gdbc6o-0 kdODGd\\"><div class=\\"styled__StCell-sc-1gdbc6o-1 cSMJMA\\">mockValue</div><div class=\\"styled__StCell-sc-1gdbc6o-1 cSMJMA\\">mockValue</div><div class=\\"styled__StCell-sc-1gdbc6o-1 cSMJMA\\">mockValue</div><div class=\\"styled__StCell-sc-1gdbc6o-1 cSMJMA\\">mockValue</div><div class=\\"styled__StCell-sc-1gdbc6o-1 cSMJMA\\">mockValue</div><span class=\\"styled__StDelete-sc-1gdbc6o-2 chhHJL\\">Ã—</span></div>"`;
